### PR TITLE
fix: support thumbnail for non-image files by upload onChange

### LINKED ISSUE #25401: Upload组件给非图片文件设置thumbUrl无效
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/eager-cdn-ifnsi?file=/index.js:768-776)

### Steps to reproduce
上传非图片的文件

### What is expected?
希望 非图片文件能在 onChange 时正常设置 thumbUrl

### What is actually happening?
非图片文件在 onChange 时设置 thumbUrl  无效

| Environment | Info |
|---|---|
| antd | 4.4.0 |
| React | ^16.12.0 |
| System | win10 |
| Browser | 版本 83.0.4103.116（正式版本） （64 位） |

### PR DESCRIPTION
**Pull request by AlbertAZ1992**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #25401

### 💡 Background and solution

`isImageUrl` call `isImageFileType` to check `file.type` first when file was uploaded manually.
so if `file.thumbUrl` was configured, skip that check.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Upload supports to show thumbnail for non-image files  as `thumbUrl` configured in `onChange` event      |
| 🇨🇳 Chinese |  Upload 支持上传非图片文件时在 `onChange` 事件中设置 `thumbUrl` 来展示缩略图       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by JadeLiuYC** [comment]
目前使用upload  组件报这个警告，需要怎么操作？    [antd: Progress] `successPercent` is deprecated. Please use `success` instead

File: components/upload/__tests__/__snapshots__/demo.test.js.snap [hunk 1]
```diff
@@ -2465,6 +2621,162 @@ exports[`renders ./components/upload/demo/picture-style.md correctly 1`] = `
...
+                        />
+                      </svg>
+                    </span>
+                  </button>
+                </span>
+              </span>
+            </div>
+          </div>
+        </span>
+      </div>
```
Comment (afc163) [inline components/upload/__tests__/__snapshots__/demo.test.js.snap:2779]: 不用修改 demo，补充到 test case 里去。

**Comment by afc163** [comment]
可以发到 master，作为一个 `修复`。

**Comment by AlbertAZ1992** [comment]
> 可以发到 master，作为一个 `修复`。

好的，稍等我改下

**Comment by AlbertAZ1992** [comment]
> 可以发到 master，作为一个 `修复`。

done

**Pull request merged**